### DISCUSSION TITLE
NextJS & i18n Localised Slugs

### DISCUSSION THREAD

**Discussion by LoyallAmsterdam** [vercel/next.js#45972] [Help]
Helo there!

I am in the finishing stage of developing a client website on my job but I do have something I can't figure out after a lot of research.  I do use Sanity CMS and the website is in multiple languages. What I am doing now is using the original pages dir, and rewrite the slugs like this:

```
source: /nl/about-us
destination: /nl/over-ons
```

Which is not always working smoothly BTW. But I did read that working with the app dir in v13, this should be solved and easier to manage with the middleware? Does anyone gave it a try already and is this indeed working? Many threads about this the last year, and it is a really important thing for a lot of businesses.

**Comment by svobik7** [comment]
I was facing the same requirement in my previous project ending up creating simple i18n routes generator that basically creates translated file routes. It is called [next-roots](https://github.com/svobik7/next-roots).

Recently I released v3 that supports Next.js 13 app folder. It might help you meet your needs.
